ast/parser.go

```diff
@@ -1147,18 +1211,18 @@ func (p *Parser) parseArray() (term *Term) {
 			}
 		}
 		// fall back to parsing as a normal array definition
-		fallthrough
-	default:
-		p.restore(s)
-		if terms := p.parseTermList(tokens.RBrack, nil); terms != nil {
-			return NewTerm(NewArray(terms...))
-		}
-		return nil
 	}
+
+	p.restore(s)
+	reset()
```

Comment:
Rationale: for the second branch here, we don't need to cache the results, since there's no third branch. (Unless this is nested, but then reset() will only decrement the counter, not disable caching.)